src/components/NcEmptyContent/NcEmptyContent.vue

```diff
@@ -121,57 +121,71 @@ export default {
 ```
 </docs>
 
+<script setup lang="ts">
+import type { Slot } from 'vue'
+
+withDefaults(defineProps<{
+	/**
+	 * A header message about an empty content shown.
+	 *
+	 * @example 'No comments'
+	 */
+	name?: string
+
+	/**
+	 * Desription of the empty content
+	 *
+	 * @example 'No comments yet, start the conversation!'
+	 */
+	description?: string
+}>(), {
+	description: '',
+	name: '',
+})
+
+defineSlots<{
+	/**
+	 * Optional slot for a button or the like
+	 */
+	action?: Slot
+
+	/**
+	 * Optional slot for adding an icon
+	 */
+	icon?: Slot
+
+	/**
+	 * Allow to add custom formatted name as an alternative to the name property.
+	 * The content passed shall be enclosed by a header element.
+	 */
+	name?: Slot
+
+	/**
+	 * Optional formatted description rendered inside a paragraph as an alternative to the description property.
+	 */
+	description?: Slot
+}>()
+</script>
+
 <template>
 	<div class="empty-content" role="note">
-		<div v-if="$slots.icon" class="empty-content__icon" aria-hidden="true">
-			<!-- @slot Optional material design icon -->
+		<div class="empty-content__icon" aria-hidden="true">
 			<slot name="icon" />
 		</div>
-		<!-- @slot Optional name if not set as property, shall be enclosed by a header element -->
 		<slot name="name">
-			<span v-if="name !== ''" class="empty-content__name">
-				{{ name }}
-			</span>
+			<span class="empty-content__name" v-text="name" />
```

Comment:
Why?

Reply:
consistent logic across the the file. But if you like 

```suggestion
			<span v-if="name !== ''" class="empty-content__name">
				{{ name }}
			</span>
```

Reply:
I'd prefer to not render unused DOM nodes even if they are hidden by CSS.

And use text interpolation to see that a node has content, but that's a nitpick